android/java/org/chromium/chrome/browser/ntp_background_images/FetchWallpaperWorkerTask.java

```diff
@@ -0,0 +1,77 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.ntp_background_images;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.util.Pair;
+import android.net.Uri;
+import java.io.InputStream;
+import java.io.FileNotFoundException;
+import android.graphics.BitmapFactory;
+
+import org.chromium.base.Log;
+import org.chromium.base.ThreadUtils;
+import org.chromium.base.ContextUtils;
+import org.chromium.base.task.AsyncTask;
+import org.chromium.chrome.browser.ntp_background_images.NTPImage;
+
+public class FetchWallpaperWorkerTask extends AsyncTask<Pair<Bitmap, Bitmap>> {
+    public interface WallpaperRetrievedCallback {
+        void bgWallpaperRetrieved(Bitmap bgWallpaper);
+        void logoRetrieved(NTPBackgroundImagesBridge.Wallpaper mWallpaper, Bitmap logoWallpaper);
+    }
+
+    private Context mContext;
+    private NTPImage mNTPImage;
+    private int mLayoutWidth;
+    private int mLayoutHeight;
+
+    // The callback to use to communicate the results.
+    private WallpaperRetrievedCallback mCallback;
+
+    public FetchWallpaperWorkerTask(NTPImage ntpImage, int layoutWidth, int layoutHeight,WallpaperRetrievedCallback callback) {
+        mNTPImage = ntpImage;
+        mLayoutWidth = layoutWidth;
+        mLayoutHeight = layoutHeight;
+        mCallback = callback;
+        mContext = ContextUtils.getApplicationContext();
+    }
+
+    @Override
+    protected Pair<Bitmap, Bitmap> doInBackground() {
+        Bitmap logoBitmap = null;
+        if (mNTPImage instanceof NTPBackgroundImagesBridge.Wallpaper) {
+            NTPBackgroundImagesBridge.Wallpaper mWallpaper = (NTPBackgroundImagesBridge.Wallpaper) mNTPImage;
+            if (mWallpaper.getLogoPath() != null ) {
+                try {
+                    Uri logoFileUri = Uri.parse("file://"+ mWallpaper.getLogoPath());
+                    InputStream inputStream = mContext.getContentResolver().openInputStream(logoFileUri);
```

Comment:
We have to close the stream below in finally block.
There are messages ` StrictMode policy violation: android.os.strictmode.LeakedClosableViolation: A resource was acquired at attached stack trace but never released.` pointing to this place.


The same for https://github.com/brave/brave-core/blob/master/android/java/org/chromium/chrome/browser/ntp_background_images/NTPUtil.java#L261 which is not a part of this PR.

Reply:
@AlexeyBarabash Good point. sorry i missed it. updating it.